### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by davidrtlake**
Pull request to merge my Siamese Network based classifier.

This library implements a Siamese Triplet Network to learn divisive embeddings for the ANDI dataset for Alzheimer’s disease. Using these embeddings, it then trains a binary classifier to classify the images.

topic-recognition branch
Commit log history
Detailed README.md file
modules.py containing the source code of the components of my model. Each component must be implemented as a class or a function.
dataset.py containing the data loader for loading and preprocessing the data.
train.py containing the source code for training, validating, testing and saving your model. The model should be imported from “modules.py” and the data loader should be imported from “dataset.py”. Make sure to plot the losses and metrics during training.
predict.py showing example usage of my trained model. Print out any results and / or provide visualisations where applicable.

Thanks,
David Lake 45330212

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- plots are provided within a folder but doesn't show up on the readme
- not a lot of information provided on the algorithm, training details
- some loss plots provided, but unclear if they are just for the training set
- minimal results shown, no accuracy plots or 2D manifolds, or prediction results

Commit messages: Good

Code: 
- good design
- well commented

Functionality/Performance: 
- classifier achieves only 71% on the test set
- not enough results/plots shown to determine triplet network performance

General comments:
- solves the problem apporpriately with a triplet network followed by a classifier, although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, plots present but not in readme, no manifold output -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, no outputs etc. -2
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
